### PR TITLE
[BugFix] Forbidden delete by complex type conditition

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/load/DeleteMgr.java [hunk 1]
```diff
@@ -508,6 +508,11 @@ private boolean checkDelete(OlapTable table, List<Partition> partitions, List<Pr
                         "is not duplicate or column type is float or double.");
             }
 
+            if (column.getType().isComplexType()) {
+                throw new DdlException(
+                        "unsupported delete condition on Array/Map/Struct type column[" + columnName + "]");
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/load/DeleteMgr.java:513]: what about json/bitmap or other complex types?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/load/DeleteMgr.java:513]: `bitmap/hll` is `Aggregate-Value` column, can't be delete condition. We can delele by `Json` type condition

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by kangkaisen — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/DeleteMgr.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by Seaven** [comment]
https://github.com/Mergifyio backport branch-2.5

**Pull request merged**